src/rules/deserializationOfUntrustedData.ts

```diff
@@ -0,0 +1,74 @@
+import { Event, EventNavigator } from '@appland/models';
+import { MatchResult, Rule, RuleLogic } from '../types';
+import { URL } from 'url';
+
+function sanitizesData(event: Event, objectId: number, label: string): boolean {
+  return (
+    event.labels.has(label) &&
+    !!event.returnValue &&
+    !!event.returnValue.object_id &&
+    event.returnValue.object_id === objectId
```

Comment:
I don't understand this requirement. Surely sanitization can return a sanitized copy of the original data?

Reply:
The idea is that the data which is deserialized needs to be trusted. We know it’s trusted if it’s the return value of a `sanitize` function. Did I get something wrong?

Reply:
Ah, ok, I understand now; took me a while :) It's a bit difficult to follow like this, although I suppose it can hardly be improved without implementing some kind of explicit data flow analysis.

BTW, maybe what confused me is that's not what the doc says -- it says sanitization should modify the data in place and return truthy. (Which is not how most sanitization functions work, so this is definitely better.)